core/src/main/java/com/linecorp/armeria/client/endpoint/AbstractDynamicEndpointGroupBuilder.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2021 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.client.endpoint;
+
+/**
+ * A skeletal builder implementation for creating other Builders that use EndpointGroup
+ */
+public abstract class AbstractDynamicEndpointGroupBuilder {
+    private boolean allowEmptyEndpoints = true;
+
+    protected AbstractDynamicEndpointGroupBuilder() { }
+
+    protected AbstractDynamicEndpointGroupBuilder(boolean allowEmptyEndpoints) {
+        this.allowEmptyEndpoints = allowEmptyEndpoints;
+    }
```

Comment:
- Please add Javadoc for the protected constructors.
- Is allowing empty endpoints the sensible default? If not, I'd recommend removing `= true` as well as the constructors.

Reply:
@trustin 
> Please add Javadoc for the protected constructors.

Yes, I will. 


> Is allowing empty endpoints the sensible default? If not, I'd recommend removing = true as well as the constructors.

I agree that `allowEmptyEndpoints =true` by default is unnecessary. I will change the logic. thanks 😄